### ISSUE TITLE
SANY returns 0 error code on Unknown Operator errors

### ISSUE THREAD

**Issue by ahelwer** [tlaplus/tlaplus#781]
## Description
When SANY parses a spec file and finds an unknown definition, it prints out an "Unknown operator" error but still returns a 0 (success) command line error code. It is important for SANY to follow command line error code conventions so it can be easily integrated with CI scripts or any other kind of script.

## Expected Behavior
I expected SANY to return a nonzero (failure) command line error code, which SANY successfully does for other types of failures like "Unknown location" errors or invalid syntax errors.

## Actual Behavior
SANY returns a 0 (success) command line error code as verified by `echo $?` in bash.

## Steps to Reproduce
1. Parse this file as Test.tla with SANY, executing the command from bash:
```tla
---- MODULE Test ----
op == DoesNotExist
====
```
2. Observe SANY outputs the Unknown operator error message
2. Run `echo $?` to get the error code of the last command executed in bash and see that it is 0

## Steps Taken to Fix
Best-effort parsing of SANY standard output to look for keywords like "error"; not ideal since different errors give different error messages and I don't know which ones I might be missing.

## Possible Fix
Ensure that SANY emits 0 ONLY if the parse succeeds fully.

## Your Environment
 - TLC version: TLC2 Version 2.17 of 02 February 2022 (rev: 3c7caa5)
 - Operating System: Arch Linux

**Comment by Calvin-L** [comment]
I believe this was fixed (?) in #483.

With the latest version of SANY you can pass `-error-codes` to have SANY exit with a nonzero result when there are errors:
```
$ cat NoSuchOperator.tla                   
---- MODULE NoSuchOperator ----
op == DoesNotExist
====

$ tla2sany -error-codes NoSuchOperator.tla

****** SANY2 Version 2.2 created 08 July 2020

Parsing file /Users/cloncari/src/tlaplus/inputs/NoSuchOperator.tla
Semantic processing of module NoSuchOperator
Semantic errors:

*** Errors: 1

line 2, col 7 to line 2, col 18 of module NoSuchOperator

Unknown operator: `DoesNotExist'.



$ echo $?
4
```

 I believe the latest 1.8.0 release has the new flag, but the 1.7.x releases (including the version you reported) do not.

Always-return-zero is an insane default, but I think the change was made as an optional flag to maintain backwards compatibility with the old (insane) behavior.

**Comment by ahelwer** [comment]
Oh wonderful, thanks! I will use that flag in my tooling. I suppose this closes the issue then, the wider issue can be rolled into https://github.com/tlaplus/tlaplus/issues/783
